### PR TITLE
rds_instance - add deletion_protection param

### LINKED ISSUE #922: rds_instance: modify delete protection
### Summary

I am trying to modify an RDS instance that has delete protection on it, preventing deletion. I would like the ability to modify this via ansible directly

```
for rds in $(aws rds describe-db-instances | jq -r '.DBInstances.[].DBInstanceIdentifier'); do
  aws rds modify-db-instance --no-deletion-protection --db-instance-identifier "${rds}"
done
```

### Issue Type

Feature Idea

### Component Name

rds_instance

### Additional Information

```yaml (paste below)
- name: Get all RDS instances
  community.aws.rds_instance_info:
    region: "{{ region }}"
    aws_access_key: "{{ aws_access_key_id }}"
    aws_secret_key: "{{ aws_secret_access_key }}"
  register: __rds_instances

- name: Remove delete protection from any DB instances
  community.aws.rds_instance:
    id: "{{ item.db_instance_identifier }}"
    deletion_protection: false
  loop: __rds_instances.instances

- name: Remove any DB instance without a final snapshot
  community.aws.rds_instance:
    id: "{{ item.db_instance_identifier }}"
    state: absent
    skip_final_snapshot: true
  loop: __rds_instances.instances

```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by jatorcasso**
Depends-On: https://github.com/ansible-collections/community.aws/pull/1116

##### SUMMARY
Fixes #922 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
rds_instance

### CODE REVIEW

File: plugins/modules/rds_instance.py [hunk 1]
```diff
@@ -666,6 +673,11 @@
   returned: always
   type: str
   sample: db-UHV3QRNWX4KB6GALCIGRML6QFA
+deletion_protection:
```
Comment (alinabuzachis) [inline plugins/modules/rds_instance.py:676]: version_added here?

File: plugins/modules/rds_instance.py [hunk 2]
```diff
@@ -666,6 +673,11 @@
...
+  description: True if the DB instance has deletion protection enabled, False if not.
```
Comment (alinabuzachis) [inline plugins/modules/rds_instance.py:677]:
```suggestion
  description: C(True) if the DB instance has deletion protection enabled, C(False) if not.
```

**Review by alinabuzachis — COMMENTED** [review]
Can you also add some integration tests please?

**Comment by jatorcasso** [comment]
recheck

**Comment by jatorcasso** [comment]
> Can you also add some integration tests please?

@alinabuzachis @markuman tests added in https://github.com/ansible-collections/community.aws/pull/1081 (since that PR included integration test changes anyway - it depends-on this PR)

**Review by markuman — APPROVED** [review]

**Review by alinabuzachis — APPROVED** [review]

**Pull request merged**